src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -931,6 +947,10 @@ private void HandleHostError(Exception exception)
                 // log file
                 NotifyInvoker(functionException.MethodName, functionException);
             }
+            else if (exception is LanguageWorkerChannelException)
+            {
+                AddLanguageWorkerChannelErrors(_functionDispatcher, FunctionErrors);
```

Comment:
The FunctionErrors collection is meant to contain function errors encountered during host startup/indexing only. E.g. say 3 functions fail to start up due to compilation errors. We retain these errors for the lifetime of the host instance so the portal can access them, etc. It shouldn't aggregate post startup runtime exceptions. Can these LanguageWorkerChannelExceptions happen after host startup as part of channel communication?

Reply:
Yes. LanguageWorkerChannelException can happen if we fail to start a process after retrying 3 times. This can happen at the start up which is most common scenario we have seen so far. It could also happen if a language process crashes at some point and we fail to restart. 
I will update the code to just write the last exception that happened. We do want to see these errors in the red balloon on portal